skcosmo/SparseMethods/Sparsified.py

```diff
@@ -0,0 +1,125 @@
+from ..preprocessing.flexible_scaler import KernelFlexibleCenterer
+from ..selection.FPS import SampleFPS
+
+
+class _Sparsified(TransformerMixin, RegressorMixin, BaseEstimator, metaclass=ABCMeta):
+    """
+    Super-class defined sparcified methods
+
+        :param kernel: the kernel used for this learning
+        :param gamma: exponential factor of the rbf and sigmoid kernel
+        :param degree: polynomial kernel degree
+        :param coef0: free term of the polynomial and sigmoid kernel
+        :param kernel_params: kernel parameter set
+        :param n_active: the size of the small dataset used in learning
+        :param regularization: regularization parameter for SparseKRR and SparseKRCovR
+        :param tol: Relative accuracy for eigenvalues (stopping criterion) The default value of 0 implies machine precision.
+        :param center: if True, centering of kernel during the learning is carried out
+        :param n_jobs: The number of jobs to use for the computation the kernel. This works by breaking down the pairwise matrix into n_jobs even slices and computing them in parallel.
+        :param n_components: number of components for which selection is carried out in SparseKPCA and SparseKRCovR
+    """
+
+    def __init__(
+        self,
+        kernel="linear",
+        gamma=None,
+        degree=3,
+        coef0=1,
+        kernel_params=None,
+        n_active=None,
+        center=True,
+        n_jobs=1,
+        selector=SampleFPS,
+    ):
+        """
+        Initializes superclass for sparse methods
+
+        :param kernel: the kernel used for this learning
+        :param gamma: exponential factor of the rbf and sigmoid kernel
+        :param degree: polynomial kernel degree
+        :param coef0: free term of the polynomial and sigmoid kernel
+        :param kernel_params: kernel parameter set
+        :param n_active: the size of the small dataset used in learning
+        :param center: if True, centering of kernel during the learning is carried out
+        :param n_jobs: The number of jobs to use for the computation the kernel. This works by breaking down the pairwise matrix into n_jobs even slices and computing them in parallel.
+        """
+        self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
+        self.kernel_params = kernel_params
+        self.n_active = n_active
+        self.center = center
+        self.n_jobs = n_jobs
+        self._selector = selector
+
+    def _project(self, A, projector):
+        """Apply a projector to matrix A
+
+        :param A: projector: string corresponding to the named projection matrix of shape (a, p)
+        :type array-like, shape (n, a)
+
+        :return: A'
+        :type array-like, shape (n, p)
+        """
+        check_is_fitted(self, projector)
+
+        A = check_array(A)
+        A_transformed = np.dot(A, self.__dict__[projector])
+        return A_transformed
+
```

Comment:
This should not be here -- not all sparse methods will conduct projection this way, and it's a simple enough operation that there's no harm in repeating.

Reply:
I agree, it's really not very complicated, but still, somehow, it's 3 lines that will have to be repeated time and time again.  I can replace it, no problem, I just think it's better to avoid unnecessary code copying.

Reply:
I'm not arguing that it won't be tedious to call, just that what defines a "Sparsified" method is the generation and manipulation of sparse kernels, _not_ the projecting of matrices. In general, I try to keep superclasses as light and focused as possible.

Reply:
I understand yes, that's fair enough. It's just that then we' d have to override this method in every inheritor class, which is also a bit weird. Is it possible to keep this method?

Reply:
I'd say omit it. There's also the point that some functions require multiple things to exist in order to call it "fitted", so that calls into question the need for the first line. I can see many use-cases overriding this, especially in the case of the covr-methods.